Feat(eos_cli_config_gen): Add support for logging format rfc5424

durd opened the pull request:
## Change Summary

This PR adds support for `logging format rfc5424`

## Related Issue(s)

Fixes #3365

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
This adds support for `logging format rfc5424`.
It adds the templates, documentation and schemas for this.

## How to test
I'm not sure how to run tests. I'm using vscode and the dev-container included in the repo.
I'm familiar with Makefiles from *BSD, but not while developing.

See separate comment below.

## Checklist

### User Checklist


### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

durd committed:
Add template for rfc5424 logging

durd committed:
Add documenation for logging format rfc5424

durd committed:
Add logging format rfc5424 to schema

durd committed:
Add logging format rfc5424 to schema

durd committed:
Add documentation for logging format rfc5424

durd commented:
I found the link to molecule in the PR checkbox comment. Looking through that I couldn't find a test for `logging format sequence-numbers`, so I figure that this change is as straight forward since I followed sequence-numbers and where it was located.
If there still is a need for a test I'll try to make one with molecule.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2:
```diff
@@ -71,6 +71,9 @@ logging format hostname ipv4
 logging format sequence-numbers
 {%     endif %}
+{%     if logging.format.rfc5424 is arista.avd.defined(true) %}
+logging format rfc5424
```
Please make sure the order is correct according to how EOS renders the config. It is probably correct already, but just confirm please :)

durd replied:
I thought my comment had made followed the resolution:
Ah, of course! I just confirmed that it wasn't in it's rendered position - I'll move it!

A question, should I change schemas to reflect this?

ClausHolbechArista replied:
Schema order does not have to match the rendered config order. The order in the schema should follow whatever makes sense for the docs.

ClausHolbechArista commented:
Moving to draft until we have tests in here. You can mark is as "ready for review" when it is ready.

For Molecule, you should add the new key to `ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/logging.yml` and run Molecule locally:
```sh
cd ansible_collections/arista/avd
molecule converge -s eos_cli_config_gen -- --limit logging
```

durd commented:
> Moving to draft until we have tests in here. You can mark is as "ready for review" when it is ready.
> 
> For Molecule, you should add the new key to `ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/logging.yml` and run Molecule locally:
> 
> ```shell
> cd ansible_collections/arista/avd
> molecule converge -s eos_cli_config_gen -- --limit logging
> ```

```
PLAY [Converge] ****************************************************************

TASK [arista.avd.eos_cli_config_gen : Verify Requirements] *********************
AVD version 4.5.0-dev1
Use -v for details.
ok: [logging -> localhost]

TASK [arista.avd.eos_cli_config_gen : Create required output directories if not present] ***
ok: [logging -> localhost] => (item=/workspaces/ansible-avd/ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/structured_configs)
ok: [logging -> localhost] => (item=/workspaces/ansible-avd/ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation)
ok: [logging -> localhost] => (item=/workspaces/ansible-avd/ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs)
ok: [logging -> localhost] => (item=/workspaces/ansible-avd/ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices)

TASK [arista.avd.eos_cli_config_gen : Include device intended structure configuration variables] ***
skipping: [logging]

TASK [arista.avd.eos_cli_config_gen : Generate eos intended configuration] *****
changed: [logging -> localhost]

TASK [arista.avd.eos_cli_config_gen : Generate device documentation] ***********
changed: [logging -> localhost]

PLAY RECAP *********************************************************************
logging                    : ok=4    changed=2    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```
Should I commit the changes? There are a lot of deletes because of `--limit logging`...

ClausHolbechArista commented:
It should not delete anything. It sometimes happens if it is the first time you run it. So try to run without limit and then commit the updates.

durd committed:
Upload artefact for issue #3366

durd commented:
Thanks, ran the command again without `--limit logging` and it recreated the cfg-files. Ran it again with `--limit` to be sure and committed and pushed the changes.

Edit: ah, sorry, I just noticed I committed the template change with the test artefacts. I hope that's ok?

durd committed:
Capitalization

ClausHolbechArista commented:
We don't care about your commits in your branch. We will squash them into a single commit using the PR title, once we merge it to the devel branch.

durd committed:
Merge branch 'aristanetworks:devel' into logging-rfc5424

durd commented:
Well, tests passed and the workflow test passed locally earlier once I corrected the capitalization in the schema.
I think I'm ready for review!

Thank you so much for your help! I couldn't have managed the last mile without your help and guidance!

ClausHolbechArista reviewed (approved):
LGTM. Thank you for this and congratulations on your first contribution to AVD! Looking forward to many more :)

tgodaA commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2:
```diff
@@ -44,6 +44,11 @@
+| RFC5424 | true |
+{%         else %}
+| RFC5424 | false |
+{%         endif %}
```
```suggestion
| RFC5424 | {{ logging.format.rfc5424 | arista.avd.default(false) }} |
```

tgodaA replied:
Apart from this minor suggestion, LGTM

carlbuchmann committed:
Update ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2

Co-authored-by: Tony Reddy Goda <tgoda@arista.com>

ClausHolbechArista commented on ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md:
```diff
@@ -53,6 +53,7 @@ interface Management1
 | Timestamp | traditional year timezone |
 | Hostname | hostname |
 | Sequence-numbers | false |
+| RFC5424 | true |
```
```suggestion
| RFC5424 | True |
```

ClausHolbechArista committed:
Update ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md

carlbuchmann reviewed (approved):
Applied review comments from @tgodaA - LGTM
